docs/integration/driver/display/st7789.rst

```diff
@@ -0,0 +1,248 @@
+=============================
+ST7789 LCD Controller driver
+=============================
+
+Overview
+-------------
+
+The `ST7789 <>`__ SOC is a popular TFT LCD controller chip made by Sitronix. It supports TFT LCD panels with a maximum resolution of 240x320 pixels, and 262k (18 bit) colors.
+It supports both serial and parallel interfaces, including 3/4 line SPI, 8080 type 8/9/16/18 bit parallel, or 6/16/18 bit RGB interfaces. The chip includes display RAM,
+various voltage and timing generators. It has a MIPI DCS compatible register set.
+
+The ST7789 LCD controller `driver <https://github.com/lvgl/lvgl/src/dev/display/st7789>`__ is a platform-agnostic driver. It implements display initialization,
+supports display rotation and implements the display flush callback. The user needs to implement only two platform-specific functions to send
+a command or pixel data to the controller via SPI or parallel bus. Typically these are implemented by calling the appropriate SDK library functions on the given platform.
+
+Prerequisites
+-------------
+
+There are no prerequisites.
+
+Configuring the driver
+----------------------
+
+Enable the ST7789 driver support in lv_conf.h, by cmake compiler define or by KConfig
+
+.. code:: c
+
+	#define LV_USE_ST7789  1
+
+Usage
+-----
+
+You need to implement two platform-dependent functions:
+
+.. code:: c
+
+	/* Send short command to the LCD. This function shall wait until the transaction finishes. */
+	int32_t my_lcd_send_cmd(lv_display_t *disp, uint8_t *cmd, size_t cmd_size, uint8_t *param, size_t param_size)
+	{
+		...
+	}
+
+	/* Send large array of pixel data to the LCD. If necessary, this function has to do the byte-swapping. This function can do the transfer in the background. */
+	int32_t my_lcd_send_color(lv_display_t *disp, uint8_t *cmd, size_t cmd_size, uint8_t *param, size_t param_size)
+	{
+		...
+	}
+
+The only difference between the ``my_lcd_send_cmd()`` and ``my_lcd_send_color()`` functions is that ``my_lcd_send_cmd()`` is used to send short commands and it is expected
```

Comment:
Can we have a send_color function like this by sending the required command before calling it?
```c
int32_t my_lcd_send_color(lv_display_t *disp, uint8_t * buf, size_t buf_size);
```

Reply:
Currently the my_lcd_send_cmd() function sends both the command and the data, and there is no way to send only a command.
We could have defined a send_cmd() and a send_data() callback, and pass a flag to send_data() to tell that the data contains pixel data, which should be sent using DMA (and maybe should be byte-swapped).
We could even pass a 'transfer_ready_cb' to send_data(), which would be used to call the lv_display_flush_ready() from the driver (currently this must be called by the user, which is not the very best solution). It is not difficult to modify the code, if you prefer this method.

Reply:
Let's leave it as it is.